scripts/localVsDissecLearning.js

```diff
@@ -0,0 +1,238 @@
+    e => !uniqueCategories.includes(e) && uniqueCategories.push(e)
+  )
+
+  const sortedOperations = allOperations.sort(
+    (a, b) =>
+      new Date(a.cozyMetadata.createdAt).valueOf() -
+      new Date(b.cozyMetadata.createdAt).valueOf()
+  )
+  const cutoffDate = new Date(
+    sortedOperations[
+      Math.round(sortedOperations.length / 2)
+    ].cozyMetadata.createdAt
+  )
+  const validationSet = sortedOperations
+    .filter(
+      e =>
+        cutoffDate.valueOf() - new Date(e.cozyMetadata.createdAt).valueOf() > 0
+    )
+    .sort((a, b) => a.id - b.id)
+  const validationIds = validationSet.map(e => e.id)
+
+  /** ===== LOCAL TRAINING ===== **/
+  const { data: localTrainingJob } = await client
+    .collection(JOBS_DOCTYPE)
+    .create('service', {
+      slug: 'dissecozy',
+      name: 'categorize',
+      pretrained: false,
+      filters: {
+        date: cutoffDate
+      }
+    })
+
+  const jobData = await client
+    .collection(JOBS_DOCTYPE)
+    .waitFor(localTrainingJob.id)
+
+  if (jobData.state !== 'done')
+    throw new Error(
+      'Local training finished with invalid status:' + jobData.state
+    )
+
+  // Measure performance
+  const locallyTrainedOperations = await client.queryAll(Q(BANK_DOCTYPE))
+  const locallyTrainedValidationSet = locallyTrainedOperations
+    .filter(e => validationIds.includes(e.id))
+    .sort((a, b) => a.id - b.id)
+
+  // Both array are sorted and contain the same elements
+  let correct = 0
+  for (let i = 0; i < validationSet.length; i++) {
+    const truth = getCategory(validationSet[i])
+    const prediction = locallyTrainedValidationSet[i].automaticCategoryId
+
+    if (truth === prediction) correct++
+  }
+
+  let localAccuracy = correct / validationSet.length
+
+  console.log('Local accuracy', localAccuracy)
+
+  /** ===== DISSEC TRAINING ===== **/
+  // Create the tree
+  // TODO: Make a dynamic tree
+  const querierWebhooks = aggregationNodes.filter(e => e.label === uri)[0]
+  const aggregatorsWebhooks = aggregationNodes.filter(e => e.label !== uri)
+  let querier = {
+    webhook: querierWebhooks.aggregationWebhook,
+    level: 0,
+    nbChild: 3,
+    aggregatorId: uuid(),
+    finalize: true
+  }
+
+  let aggregators = [
+    {
+      webhook: aggregatorsWebhooks[0].aggregationWebhook,
+      level: 1,
+      nbChild: 6,
+      parent: querier,
+      aggregatorId: uuid(),
+      finalize: false
+    },
+    {
+      webhook: aggregatorsWebhooks[1].aggregationWebhook,
+      level: 1,
+      nbChild: 6,
+      parent: querier,
+      aggregatorId: uuid(),
+      finalize: false
+    },
+    {
+      webhook: aggregatorsWebhooks[2].aggregationWebhook,
+      level: 1,
+      nbChild: 6,
+      parent: querier,
+      aggregatorId: uuid(),
+      finalize: false
+    }
+  ]
+
+  // All instances with data will contribute
+  let contributors = aggregatorsWebhooks.map(e => ({
+    ...e,
+    level: 2,
+    nbChild: 0,
+    parents: aggregators
+  }))
+
```

Comment:
This would probably ease the code readability to put the tree creation into a function